node/src/components/network/conman.rs

```diff
@@ -1006,6 +1095,33 @@ where
     }
 }
 
+/// Calculates a sensible do-not-call-timeout from a given error.
+fn reconnect_delay_from_rpc_server_error(cfg: &Config, err: &RpcServerError) -> Duration {
+    let Some((header, raw)) = err.as_remote_other_err() else {
+        return cfg.significant_error_backoff.into();
+    };
+
+    if !header.is_error() || header.error_kind() != ErrorKind::Other {
+        return cfg.significant_error_backoff.into();
+    }
+
+    // It's a valid user error with a payload.
+    let Some(peer_err) = PeerError::deserialize(raw) else {
+        rate_limited!(RPC_ERROR_OTHER_INVALID_MESSAGE, |dropped| warn!(
+            dropped,
+            "failed to deserialize a custom error message"
+        ));
+        return cfg.significant_error_backoff.into();
+    };
+
+    match peer_err {
+        PeerError::YouAreBanned { time_left, .. } => {
+            time_left.min(cfg.permanent_error_backoff.into())
```

Comment:
Shouldn't this be `max`?  If not, shouldn't we just use `time_left`, since presumably if we try to reconnect earlier than `time_left`, the peer will reject the connection attempt?

Reply:
No, we want to connect as soon permitted, which is after the ban. The assumption is that the `permanent_error_backoff` is always going to be "very large", i.e. larger than a ban. The check is to defend against someone filling up our do-not-call list with ludicrously high timeouts.

Reply:
We already have an upper limit on the `do_not_call` list don't we?